### ISSUE TITLE
Sharing notebooks with users

### ISSUE THREAD

**Issue by biojerm** [jupyterhub/the-littlest-jupyterhub#144]
To prevent a [XY problem](https://en.wikipedia.org/wiki/XY_problem), I am going to describe what I would like to do, then explain what I see as my current issue.  Please feel free to point me in another direction or to some other tool if you think it would work better, I am pretty new at this.

I am trying to use TLJH and JupyterNotebooks to distribute python tools to my co-workers.  I plan on being the main developer of the notebooks, then have my co-workers log into the TLJH server (currently local) to grab an instance of the most up to date set of notebooks and use the tools.  I envision a few people with admin access who could make changes to the notebooks, then a lot more people who could use but not overwrite the notebooks. 

The issues I am trying to solve:  
1) How do I point TLJH to a folder with the notebooks I would like to share (this would be a folder on the server instance I assume)? Right now I only see examples.desktop in the tree when a new instance is created.  I saw #106 asked a similar question.   

2) How do I make it so that when a user logs in, a 'fresh' set notebook is created and served with the most recent notebooks from the folder in question 1? I saw the page on [nbgitpuller](https://the-littlest-jupyterhub.readthedocs.io/en/latest/howto/content/nbgitpuller.html), but I would prefer not to require the users to regularly check if  their version is the latest.  I would rather have a fresh instance of the notebooks every time.

P.S.  Is there a better place for me to bring up these issues? I don't want to pollute the issue boards with [RTFM](https://en.wikipedia.org/wiki/RTFM), and newbie questions.

**Comment by betatim** [comment]
You could [create a symbolic link](https://the-littlest-jupyterhub.readthedocs.io/en/latest/howto/content/share-data.html#option-2-create-a-read-only-shared-folder-for-data) to the directory in which the "master" copies are on the same machine.

I'm wondering if it is possible to run `nbgitpuller` every time a user logs in to the JupyterHub. Does one of the shell profile files (`.bashrc`, `.profile`, ..`) get executed when a user logs in via the hub? Then you could add the nbgitpuller command there and it would run automatically when a user logs is. Then combine that with the [`/etc/skel`](https://the-littlest-jupyterhub.readthedocs.io/en/latest/howto/content/share-data.html#add-a-link-to-the-shared-folder-in-the-user-home-directory) idea for automatically creating files in users home directories.

**Comment by fm75** [comment]
@biojerm Did it work? I am not sure what you want from the fresh notebook. I assume, maybe wrongly, that you want your co-workers to be able to run the notebooks. If they run and save the notebook, do you want to overwrite it whenever you want a replacement?

If you want each to start with **your** latest and greatest, perhaps [mybinder](https://mybinder.org/) might be a better approach. If you do that, then you don't even need your own jupyterhub. Just storing your notebooks on GitHub would be sufficient. But your co-workers work is temporary unless they make their own copy.

**Comment by biojerm** [comment]
@fm75 I ended up using a notebook with a `nbgitpuller` link to update the user notebooks.  For my current needs `nbgitpuller` works fine.  My desire to have 'fresh' notebooks each time was really unnecessary with the reality of everyone's workflows.  Plus, people being able to save their work to their account ended up being a nice feature.   To make sure everyone had a `nbgitpuller` link when a tljh instance first spawned I ended up putting a notebook with the `nbgitpuller` link into the default directory that is copied when a tljh instance is created.  I think it was `/etc/skele` or something like that, I will update this comment in the morning with the correct path when I am back at work.  

Thanks for the suggestion about 'mybinder'  I will check it out.
